lib/preprocess.js

```diff
@@ -89,15 +92,37 @@ export const preprocess = async ({
   })
 }
 
-export const fetchMeasurementsViaClient = async (web3Storage, cid) => {
-  const res = await web3Storage.get(cid)
-  const files = await res.files()
-  const measurements = JSON.parse(await files[0].text())
-  return measurements
+export const fetchMeasurementsTrustless = async cid => {
+  const res = await fetch(
+    `https://${encodeURIComponent(cid)}.ipfs.w3s.link?format=car`
+  )
+  if (!res.ok) {
+    const msg = `Cannot fetch measurements ${cid}: ${res.status}\n${await res.text()}`
+    throw new Error(msg)
+  }
+  const reader = await CarReader.fromIterable(res.body)
+  const entries = exporter(cid, {
+    async get (cid) {
+      const block = await reader.get(cid)
+      await validateBlock(block)
+      return block.bytes
+    }
+  })
+  for await (const entry of entries) {
+    if (entry.type === 'file' || entry.type === 'raw') {
```

Comment:
I find this suspicious. Under which conditions will we receive `file`, and when will we receive `raw` instead? Isn't it enough to have only one condition?

Reply:
This came up from code review. Depending on the file size of the entry, it will either be plain text or binary encoded:

https://github.com/web3-storage/w3up/blob/e8bffe2ee0d3a59a977d2c4b7efe425699424e19/packages/upload-client/src/unixfs.js#L11